src/components/CreateNewProjectWizard/MainPane.js

```diff
@@ -29,29 +31,116 @@ type Props = {
   handleBlurProjectName = () => this.props.focusField(null);
+  handleFocusStarter = () => this.props.focusField('projectStarter');
 
   updateProjectName = (projectName: string) =>
     this.props.updateFieldValue('projectName', projectName);
   updateProjectType = (projectType: ProjectType) =>
     this.props.updateFieldValue('projectType', projectType);
   updateProjectIcon = (projectIcon: string) =>
     this.props.updateFieldValue('projectIcon', projectIcon);
+  updateGatsbyStarter = (selectedStarter: string) =>
+    this.props.updateFieldValue('projectStarter', selectedStarter);
 
+  projectSpecificSteps() {
+    const { activeField, projectType, projectStarter } = this.props;
+    switch (projectType) {
+      case 'gatsby':
+        return (
+          <FadeIn key="step-starter">
+            <FormField
+              label="Project Starter"
+              isFocused={activeField === 'projectStarter'}
+            >
+              <ProjectStarterSelection
+                isFocused={activeField === 'projectStarter'}
+                handleFocus={this.handleFocusStarter}
+                onSelect={this.updateGatsbyStarter}
+                projectStarter={projectStarter}
+              />
+            </FormField>
+          </FadeIn>
+        );
+      default:
+        return null;
+    }
+  }
+
+  renderConditionalSteps(currentStepIndex: number) {
+    const { activeField, projectType, projectIcon } = this.props;
+    const buildSteps: Array<?React$Node> = [
+      // currentStepIndex = 0
+      <FadeIn key="step-type">
+        <FormField
+          label="Project Type"
+          isFocused={activeField === 'projectType'}
+        >
+          <ProjectTypeSelection
+            projectType={projectType}
+            onProjectTypeSelect={selectedProjectType =>
+              this.updateProjectType(selectedProjectType)
+            }
+          />
+        </FormField>
+      </FadeIn>,
+      this.projectSpecificSteps(), // currentStepIndex = 1
+      <FadeIn key="step-icon">
+        <FormField
+          label="Project Icon"
+          focusOnClick={false}
+          isFocused={activeField === 'projectIcon'}
+        >
+          <ProjectIconSelection
+            selectedIcon={projectIcon}
+            randomize={true}
+            limitTo={8}
+            onSelectIcon={this.updateProjectIcon}
+          />
+        </FormField>
+      </FadeIn>,
+    ].filter(step => !!step);
+
+    const renderedSteps: Array<?React$Node> = buildSteps.slice(
+      0,
+      currentStepIndex
+    );
+
+    return {
+      lastIndex: buildSteps.length,
+      steps: renderedSteps,
+    };
+  }
+  validateField(currentStepIndex: number, lastIndex: number) {
+    // No validation for projectStarter as it is optional
+    const { projectIcon, projectType } = this.props;
+
+    return (
+      (currentStepIndex > 0 && !projectType) ||
+      (currentStepIndex >= lastIndex && !projectIcon)
+    );
```

Comment:
I'd love for this to be broken into a couple steps, so that the steps can be given variable names to convey meaning.

```js
const needsProjectType = !projectType && currentStepIndex > 0;
const needsProjectIcon = !projectIcon && currentStepIndex >= lastIndex;

return needsProjectType || needsProjectIcon;
```

Also, I was initially confused because I assumed `validateField` would return `true` if the field was invalid... how about `isSubmitDisabled` as the name?

Reply:
Good idea. Renaming to `isSubmitDisabled` makes it easier to understand. Splitting is also OK as it improves readability. 
I first though that it's clear what these checks are doing but for new devs it's better to have that additional naming.